fastlane/lanes/localization.rb

```diff
@@ -209,70 +166,47 @@
-    push_to_git_remote
-  end
-
-  desc "Downloads Jetpack's translated metadata from GlotPress"
-  lane :download_jetpack_metadata_strings do |options|
-    UI.message('Hey')
-    app_values = APP_SPECIFIC_VALUES[:jetpack]
-    values = options[:version].split('.')
-    files = {
-      "release_note_#{values[0]}#{values[1]}" => { desc: "changelogs/#{options[:build_number]}.txt", max_size: 500, alternate_key: "release_note_short_#{values[0]}#{values[1]}" },
-      play_store_app_title: { desc: 'title.txt', max_size: 30 },
-      play_store_promo: { desc: 'short_description.txt', max_size: 80 },
-      play_store_desc: { desc: 'full_description.txt', max_size: 4000 }
-    }
-
-    delete_old_changelogs(app: 'jetpack', build: options[:build_number])
-    download_path = File.join(Dir.pwd, app_values[:metadata_dir], 'android')
-    gp_downloadmetadata(
-      project_url: app_values[:glotpress_metadata_project],
-      target_files: files,
-      locales: JP_RELEASE_NOTES_LOCALES,
-      download_path: download_path
-    )
-
-    # For WordPress, the en-US release notes come from using the source keys (instead of translations) downloaded from GlotPress' en-gb locale (which is unused otherwise).
-    # But for Jetpack, we don't have an unused locale like en-gb in the GP release notes project, so copy from source instead as a fallback
-    metadata_source_dir = File.join(Dir.pwd, '..', 'WordPress', 'jetpack_metadata')
-    FileUtils.cp(File.join(metadata_source_dir, 'release_notes.txt'), File.join(download_path, 'en-US', 'changelogs', "#{options[:build_number]}.txt"))
-    FileUtils.cp(
-      ['title.txt', 'short_description.txt', 'full_description.txt'].map { |f| File.join(metadata_source_dir, f) },
-      File.join(download_path, 'en-US')
-    )
+    version = options.fetch(:version, android_get_app_version)
+    build_number = options.fetch(:build_number, android_get_release_version['code'])
+
+    # If no `app:` is specified, call this for both WordPress and Jetpack
+    apps = options[:app].nil? ? %i[wordpress jetpack] : Array(options[:app]&.downcase&.to_sym)
+
+    apps.each do |app|
+      app_values = APP_SPECIFIC_VALUES[app]
+
+      version_suffix = version.split('.').join
+      files = {
+        "release_note_#{version_suffix}" => { desc: "changelogs/#{build_number}.txt", max_size: 500, alternate_key: "release_note_short_#{version_suffix}" },
+        play_store_app_title: { desc: 'title.txt', max_size: 30 },
+        play_store_promo: { desc: 'short_description.txt', max_size: 80 },
+        play_store_desc: { desc: 'full_description.txt', max_size: 4000 }
+      }
+
+      delete_old_changelogs(app: app, build: build_number)
+
+      download_path = File.join(Dir.pwd, app_values[:metadata_dir], 'android')
+      locales = { wordpress: WP_RELEASE_NOTES_LOCALES, jetpack: JP_RELEASE_NOTES_LOCALES }[app]
```

Comment:
Ideally this info should go into `APP_SPECIFIC_VALUES` constant like the others.

But for this to be possible, I'd need to first finish my long-postponed work on adding a Locale Helper to the `release-toolkit` (initially started in https://github.com/wordpress-mobile/release-toolkit/pull/296, and re-implemented a bit differently and more simply recently as part of https://github.com/wordpress-mobile/release-toolkit/pull/401/files#diff-bdb20f98842eb1bfe87ab2b48e33f15dccb6779e9509cf514759d8bbaa605378, both still being in Draft stage), so that we could just use the list of GlotPress locale codes in new `app_locales:` and `release_notes_locales:` keys of the `APP_SPECIFIC_LOCALES` constant — instead of having to move all the long and complex constants like `ALL_LOCALES` and all the related `*_LOCALES_*`constants defined next to it and doing some dance based on it like we do, which makes it quite impractical to move to `APP_SPECIFIC_LOCALES` in `Fastfile`.

So I thought that this solution for this line would do in the meantime; I really hope I can continue and finish my work on the Locale Helper in `release-toolkit` at some point, which would then simplify a _lot_ of code manipulating locale-related info and constants in many places in our Fastfile code, including this here.

Reply:
> I really hope I can continue and finish my work on the Locale Helper in release-toolkit at some point

🤞 ❤️ 🙏